sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobCheckpointStoreInternal.cs

```diff
@@ -441,10 +441,10 @@ await foreach (BlobItem blob in ContainerClient.GetBlobsAsync(traits: BlobTraits
                 offset = result;
                 startingPosition = EventPosition.FromOffset(result, false);
             }
-            else if (metadata.TryGetValue(BlobMetadataKey.SequenceNumber, out str) && long.TryParse(str, NumberStyles.Integer, CultureInfo.InvariantCulture, out result))
+            if (metadata.TryGetValue(BlobMetadataKey.SequenceNumber, out str) && long.TryParse(str, NumberStyles.Integer, CultureInfo.InvariantCulture, out result))
             {
                 sequenceNumber = result;
-                startingPosition = EventPosition.FromSequenceNumber(result, false);
+                startingPosition ??= EventPosition.FromSequenceNumber(result, false);
```

Comment:
This maintains the existing behavior where offset takes precedence over sequence number when creating the position.

Reply:
This is consistent with the expected behavior of the Event Processor as well.